pkg/server/devfile-handler.go

```diff
@@ -50,33 +45,53 @@ func (s *Server) devfileHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	containerComponents := devfileObj.Data.GetDevfileContainerComponents() //TODO: filter thru Console attribute, right now if there is more than one component container err out
+	filterOptions := common.DevfileOptions{
+		Filter: map[string]interface{}{
+			"tool": "console-import",
+		},
+	}
+
+	containerComponents, err := devfileObj.Data.GetDevfileContainerComponents(filterOptions) //For Dev Preview, if there is more than one component container err out
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to get the container component from devfile with attribute 'tool: console-import': %v", err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusBadRequest, serverutils.ApiError{Err: errMsg})
+		return
+	}
 	if len(containerComponents) != 1 {
 		errMsg := "Console Devfile Import Dev Preview, supports only one component container"
 		klog.Error(errMsg)
 		serverutils.SendResponse(w, http.StatusBadRequest, serverutils.ApiError{Err: errMsg})
 		return
 	}
 
-	deploymentResource, err := getDeployResource(data, devfileObj)
+	deploymentResource, err := getDeployResource(data, devfileObj, filterOptions)
 	if err != nil {
 		errMsg := fmt.Sprintf("Failed to get deployment resource for the devfile: %v", err)
 		klog.Error(errMsg)
 		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: errMsg})
 		return
 	}
 
-	service, err := getService(devfileObj)
+	service, err := getService(devfileObj, filterOptions)
 	if err != nil {
 		errMsg := fmt.Sprintf("Failed to get service for the devfile: %v", err)
 		klog.Error(errMsg)
 		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: errMsg})
 		return
 	}
 
+	dockerfilePath := devfileObj.Data.GetMetadata().Attributes.GetString("alpha.build-dockerfile", &err)
```

Comment:
@maysunfaisal -- not a Go expert by any means... but is the `err` not supposed to be a 2nd argument here? Or is the `&err` passing in a pointer that is updated inside?

```suggestion
	dockerfilePath, err := devfileObj.Data.GetMetadata().Attributes.GetString("alpha.build-dockerfile", &err)
```

Reply:
That is correct. The err pointer is passed and updated if there is any err. Thats how the go module func was written.

Reply:
We should also document the fact that dockerfile location is to be provided by attribute `"alpha.build-dockerfile"` - https://github.com/maysunfaisal/node-bulletin-board-3/blob/main/devfile.yaml#L6 @andrewballantyne @serenamarie125

Reply:
Good point @maysunfaisal -- The fallback is co-located with the `devfile.yaml`, correct?

Reply:
Well atm, if its not available, we err out by the pointer you mentioned..

![Screen Shot 2020-12-04 at 2 53 14 PM](https://user-images.githubusercontent.com/31771087/101208730-97f2a300-3640-11eb-8db1-ab6b35fe4f96.png)

Reply:
Good to note @maysunfaisal... @serenamarie125 this will be another required thing for them